Sources/TokamakCore/App/AppStorage.swift

```diff
@@ -0,0 +1,197 @@
+}
+
+@propertyWrapper public struct AppStorage<Value>: ObservedProperty {
+  let provider: _AppStorageProvider?
+  @Environment(\._defaultAppStorage) var defaultProvider: _AppStorageProvider?
+  var unwrappedProvider: _AppStorageProvider {
+    provider ?? defaultProvider!
+  }
+
+  let key: String
+  let defaultValue: Value
+  let wrapValue: (Value) -> String
+  let unwrapValue: (String) -> Value?
+
+  var objectWillChange: AnyPublisher<(), Never> {
+    unwrappedProvider.publisher.eraseToAnyPublisher()
+  }
+
+  public var wrappedValue: Value {
+    get {
+      if let stringValue = unwrappedProvider.read(key: key) {
+        return unwrapValue(stringValue) ?? defaultValue
+      }
+      return defaultValue
+    }
+    nonmutating set {
+      unwrappedProvider.store(key: key, value: wrapValue(newValue))
+    }
+  }
+
+  public var projectedValue: Binding<Value> {
+    .init {
+      wrappedValue
+    } set: {
+      wrappedValue = $0
+    }
+  }
+}
+
+extension AppStorage: DynamicProperty {}
+
+extension AppStorage {
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil) where Value == Bool {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { String($0) }
+    unwrapValue = { Bool($0) }
+  }
+
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil) where Value == Int {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { String($0) }
+    unwrapValue = { Int($0) }
+  }
+
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil) where Value == Double {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { String($0) }
+    unwrapValue = { Double($0) }
+  }
+
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil) where Value == String {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { $0 }
+    unwrapValue = { $0 }
+  }
+
+  public init(wrappedValue: Value,
+              _ key: String,
+              store: _AppStorageProvider? = nil)
+    where Value: RawRepresentable, Value.RawValue == Int {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { String($0.rawValue) }
+    unwrapValue = {
+      guard let rawValue = Int($0) else {
+        return nil
+      }
+      return Value(rawValue: rawValue)
+    }
+  }
+
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil)
+    where Value: RawRepresentable, Value.RawValue == String {
+    defaultValue = wrappedValue
+    self.key = key
+    provider = store
+    wrapValue = { $0.rawValue }
+    unwrapValue = { Value(rawValue: $0) }
+  }
+}
+
+extension AppStorage where Value: ExpressibleByNilLiteral {
+  public init(wrappedValue: Value, _ key: String, store: _AppStorageProvider? = nil) where Value == Bool? {
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/TokamakCore/App/AppStorage.swift;
  Line: 126;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 107 characters (`line_length`)